src/toil/job.py

```diff
@@ -429,41 +429,45 @@ def asyncWrite():
 
         def getLocalTempDir(self):
             """
-            Get a new local temporary directory. This directory will exist for the 
+            Get the absolute path to a new local temporary directory. 
+            This directory will exist for the 
             duration of the job only, and is guaranteed to be deleted once
             the job terminates, removing all files it contains recursively. 
             """
-            return tempfile.mkdtemp(prefix="t", dir=self.localTempDir)
+            return os.path.abspath(tempfile.mkdtemp(prefix="t", dir=self.localTempDir))
         
         def getLocalTempFile(self):
             """
-            Get a local temporary file. This file will exist for the duration of the job only, and
+            Get an absolute path to a local temporary file. 
+            This file will exist for the duration of the job only, and
             is guaranteed to be deleted once the job terminates.
             """
             handle, tmpFile = tempfile.mkstemp(prefix="tmp", 
                                                suffix=".tmp", dir=self.localTempDir)
             os.close(handle)
-            return tmpFile
+            return os.path.abspath(tmpFile)
```

Comment:
Isn't this sensitive to the current working directory? If so, that would be bad.